Library/Homebrew/dev-cmd/determine-cask-runners.rb

```diff
@@ -0,0 +1,353 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "abstract_command"
+require "tap"
+require "utils/github/api"
+require "cli/parser"
+require "system_command"
+
+module Homebrew
+  module DevCmd
+    class DetermineCaskRunners < AbstractCommand
+      MAX_JOBS = 256
+
+      # Weight for each arch must add up to 1.0.
+      INTEL_RUNNERS = T.let({
+        { symbol: :monterey, name: "macos-12", arch: :intel } => 0.0,
+        { symbol: :ventura,  name: "macos-13", arch: :intel } => 1.0,
+      }.freeze, T::Hash[T::Hash[Symbol, Symbol], Float])
+      ARM_RUNNERS = T.let({
+        { symbol: :sonoma,   name: "macos-14", arch: :arm   } => 0.0,
+        { symbol: :sequoia, name: "macos-15", arch: :arm }    => 1.0,
+      }.freeze, T::Hash[T::Hash[Symbol, Symbol], Float])
+      RUNNERS = T.let(INTEL_RUNNERS.merge(ARM_RUNNERS).freeze,
+                      T::Hash[T::Hash[Symbol, Symbol], Float])
+
+      cmd_args do
+        description <<~EOS
+          Generate a GitHub Actions matrix for a given pull request URL or list of cask names.
+          For internal use in Homebrew taps.
+        EOS
+
+        flag "--url=",
+             description: "URL of a pull request to generate a matrix for."
+        flag "--tap=",
+             description: "The tap."
+        switch "--skip-install",
+               description: "Skip installing casks"
+        switch "--new",
+               description: "Run new cask checks"
+        switch "--syntax-only",
+               description: "Only run syntax checks"
+
+        conflicts "--url", "--casks"
+        conflicts "--syntax-only", "--url"
+        conflicts "--syntax-only", "--casks"
+        conflicts "--syntax-only", "--skip-install"
+        conflicts "--syntax-only", "--new"
+
+        named_args [:cask], min: 0
```

Comment:
Can this be

```suggestion
        named_args [:cask, :tap, :url], min: 1
```

or

```suggestion
        named_args [:cask, :tap], min: 0
```

?

Reply:
I went with `named_args [:cask, :url]` and moved to using a switch to choose which is being provided.
I also dropped the tap argument and this should always be run through a GitHub action where the repository is fetched from the provided environment variable.